mlflow/tensorflow/__init__.py

```diff
@@ -770,6 +803,46 @@ def train(original, self, *args, **kwargs):
 
         result = original(self, *args, **kwargs)
 
+        def _extract_data_from_input_fn():
+            """
+            Extracts sample data from dict (str -> ndarray) or Tensor type.
+
+            :return: a slice (of limit `INPUT_EXAMPLE_SAMPLE_ROWS`)
+                     of the input of type `np.ndarray`
+            """
+
+            def _extract_dict_or_tensor(next_input):
+                input_ex_dict_tensor = None
+                if isinstance(next_input, dict):
+                    input_ex_dict_tensor = {
+                        k: _extract_tensor_as_numpy_input_example_slice(v)
+                        for k, v in next_input.items()
+                    }
+                elif isinstance(next_input, tensorflow.Tensor):
+                    input_ex_dict_tensor = _extract_tensor_as_numpy_input_example_slice(next_input)
+                return input_ex_dict_tensor
+
+            input_training_data = kwargs.get("input_fn")()
+            input_features = None
+            if isinstance(input_training_data, tuple):
+                input_features = _extract_dict_or_tensor(input_training_data[0])
+            elif isinstance(input_training_data, tensorflow.data.Dataset):
+                train_iter = iter(input_training_data)
+                next_input = next(train_iter)[0]
+                input_features = _extract_dict_or_tensor(next_input)
+            else:
+                warnings.warn(
+                    "Tensorflow estimator doesn't support this "
+                    f"type for training: {type(input_training_data)}."
+                )
```

Comment:
Can we update this message to make it clear that this type isn't support for *input example logging*. The rest of TensorFlow autologging capabilities will continue to work, so we don't want to confuse users.